doc/content/examples/example4.md

```diff
@@ -34,4 +31,72 @@ layer effectively simulates the interface between two elastic layers (see [fig:d
+
+!listing examples/ex04/ex04b/2BlockFriction_Isoilunit.i
+
+For this example, the units defining the parameters are N, kg, and meter (m); and the material definition is created using the following input:
+
+!listing examples/ex04/ex04b/2BlockFriction_Isoilunit.i
+ start=Materials
+ end=Preconditioning
+
+where;
+
+- `soil type = 'user-defined'` specifies that the backbone curve will be defined by the user and will correspond to a Coulomb friction formulation.
+
+- `backbone curve = backbone_curveunit.csv` The backbone curve is used to define critical shear stress at the interface which is obtained by multiplying the user-defined friction coefficient with the normal stresses . The shear strain is obtained by dividing the frictional shear stress with the shear modulus of the concrete. The above mentioned stress-strain values are entered via a .csv file or can be defined directly. The initial shear modulus of the thin layer is the same as that of concrete. The motivation is to model the contact element as a set of springs which yields once the critical stresses are reached.
+
+- `initial_shear_modulus = '2.71e10'` specifies the small strain shear modulus (elastic property) of
+ the soil and is equivalent to the shear modulus of the concrete block. It should be noted that if we were to model an embedded foundation the shear modulus of the thin layer must match that of the surrounding soil strata.
+
+- `poissons_ratio = '0.15'` specifies the Poisson's ratio for the thin layer elements (same as concrete). The linear elastic properties are matched with concrete in order to ensure uniform scaling of stresses during failure.
+
+- `block = 1002` specifies that the material will be assigned to a mesh volume with block id = 1002.
+
+- `initial_stress = 'initial_ymid 0 0 0 initial_ymid 0 0 0 initial_zzmid` see [Getting Started](manuals/user/index.md). This is not needed if pressure dependency is set to true or when the initial stresses are unknown.
+
+- `density = '8792'` specifies the density of the thin layer which is the same as concrete.
+
+- `p_ref = '904918'` In this example, the reference pressure is the hydrostatic stress on the thin layer. If the normal stress cannot be easily estimated, as is the case for this concrete block resting on top of the thin layer, the reference pressure can be obtained by performing a gravity analysis by switching off the inertia kernels for the first few time steps.
+
+- `pressure dependency = true` This ensures that the stress-strain behavior of the interface element scales linearly with the normal pressure on the interface. When the confining pressure (due to gravity) on the interface changes, the shear strength of the interface element changes as well, as it is a function of the normal pressure. The value of $a_0$, $a_1$ and $a_2$ are set to 0, 0, 1 respectively in order to completely activate the pressure dependency.
+
```

Comment:
Setting $a_0$, $a_1$ and $a_2$ only varies yield strength with pressure to model the Coulomb friction type stress-strain behavior. To model gapping between soil and the concrete (which is not modeled in this current test), $b_exp$ should be non-zero so that the shear modulus decreases to zero when the thin layer element is under tension. When $b_exp$ is zero, the shear modulus is the same in tension or compression, which is unrealistic for contact scenarios.